### PR TITLE
Filter VINs

### PR DESCRIPTION
**Pull request by bassrock**
This adds in support to only add specific Teslas (or none) to homeassistant if you have multiple on your account. For my particular setup I have 2 energy sites and a tesla on my account. However I did not want the tesla added to my homeassistant and getting polled.

## Implementation Details
* I tried to add new tests, but they were failing locally with an error about not finding the tesla integration. I tried to solve this but could not find much information online about how to. Would appreciate any thoughts. 
* The default value for including all teslas is to use a space, this is because the homeassistant options flow will not pass through an empty value to the options flow. So if you add in a list of vins, then decide to remove them, the options flow will not signify that the vin value changed. So you have to use a space to signify this.   Ideally including all vins should be a checkbox and if not checked the text box appears, but I could not find out how to do this in the homeassistant documentation.

## Upgrades 
I did have to upgrade the teslajsonpy integration and the homeassistant integration because of two new features:
* Teslajsonpy - https://github.com/zabuldon/teslajsonpy/pull/263
* Homeassistant - https://github.com/home-assistant/core/pull/49912

Marking this as a draft because of the tests and some uncertainty. I will note that I am using this version of the integration as a daily driver just fine.

### CODE REVIEW

**Comment by alandtse** [comment]
What's the error you're seeing on local tests? Hard to provide advice without more details. Also, I'm not enthused by requiring a space for all as it will generate issues. HA hasn't implemented select lists yet?

**Comment by Olen** [comment]
I also get the pytest-errors about missing integration:

ERROR    homeassistant.config_entries:config_entries.py:711 Cannot find integration tesla_custom

It happens on 7 of 10 tests in tests/test_config_flow.py and I guess the problem is this:

```
    with patch(
        "custom_components.tesla_custom.config_flow.TeslaAPI.connect",
        return_value={
            "refresh_token": TEST_TOKEN,
            CONF_ACCESS_TOKEN: TEST_ACCESS_TOKEN,
            CONF_EXPIRATION: TEST_VALID_EXPIRATION,
        },
    ):
```

The 3 test_option_flow... tests completes successfully.

**Comment by Olen** [comment]
And I think you can use a multi_select in config_flow (however, I have not tested it).

It should be something like:
```
vol.Optional(
    CONF_VINS_TO_INCLUDE, default=DEFAULT_VINS_TO_INCLUDE
): cv.multi_select(ALL_VINS),
```
(Not tested and verified against your code)

File: pyproject.toml [hunk 1]
```diff
@@ -7,11 +7,11 @@ license = "Apache-2.0"
...
 [tool.poetry.dependencies]
 python = "^3.9"
-teslajsonpy = "^1.4.1"
+teslajsonpy = "^1.4.2"
 
 [tool.poetry.dev-dependencies]
-homeassistant = "^2021.3.4"
-pytest-homeassistant-custom-component = "~=0.3.1"
+homeassistant = "=2021.6.0b0"
+pytest-homeassistant-custom-component = "=0.4.0"
```
Comment (alandtse) [inline pyproject.toml:14]: I wonder if this change is what is breaking pytest.

Comment (bassrock) [reply pyproject.toml:14]: LIkely. I had to update to at least that version of homeassistant though for this function https://github.com/home-assistant/core/pull/49912

Comment (alandtse) [reply pyproject.toml:14]: BTW, this was confirmed to be the cause. https://github.com/alandtse/tesla/issues/170

Comment (alandtse) [reply pyproject.toml:14]: As a note, I figured out the fix if you ever want to pick this up again.

Comment (bassrock) [reply pyproject.toml:14]: @alandtse I actually closed this because it looks like this was kinda implemented in the latest version with the Poll switch.

**Comment by alandtse** [comment]
Oh. I didn't realize you didn't know about the polling switches. Those have been available for a long time. I thought you wanted some way to filter during the device discovery.

**Pull request closed without merging**